### PR TITLE
Use templates on all ITIL objects

### PR DESCRIPTION
**Pull request by trasher**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

### CODE REVIEW

File: front/changetemplatehiddenfield.form.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+$itiltype = 'Change';
+$fieldtype . 'Hidden';
```
Comment (cedric-anne) [inline front/changetemplatehiddenfield.form.php]:
```suggestion
$fieldtype = 'Hidden';
```

File: front/changetemplatemandatoryfield.form.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+$itiltype = 'Change';
+$fieldtype . 'Mandatory';
```
Comment (cedric-anne) [inline front/changetemplatemandatoryfield.form.php]:
```suggestion
$fieldtype = 'Mandatory';
```

File: front/changetemplatepredefinedfield.form.php [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+$itiltype = 'Change';
+$fieldtype . 'Predefined';
```
Comment (cedric-anne) [inline front/changetemplatepredefinedfield.form.php]:
```suggestion
$fieldtype = 'Predefined';
```

File: front/problemtemplatehiddenfield.form.php [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+$itiltype = 'Problem';
+$fieldtype . 'Hidden';
```
Comment (cedric-anne) [inline front/problemtemplatehiddenfield.form.php]:
```suggestion
$fieldtype = 'Hidden';
```

File: front/problemtemplatemandatoryfield.form.php [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+$itiltype = 'Problem';
+$fieldtype . 'Mandatory';
```
Comment (cedric-anne) [inline front/problemtemplatemandatoryfield.form.php]:
```suggestion
$fieldtype = 'Mandatory';
```

File: front/problemtemplatepredefinedfield.form.php [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+$itiltype = 'Problem';
+$fieldtype . 'Predefined';
```
Comment (cedric-anne) [inline front/problemtemplatepredefinedfield.form.php]:
```suggestion
$fieldtype = 'Predefined';
```

File: front/tickettemplatehiddenfield.form.php [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+
+$itiltype = 'Ticket';
+$fieldtype . 'Hidden';
```
Comment (cedric-anne) [inline front/tickettemplatehiddenfield.form.php]:
```suggestion
$fieldtype = 'Hidden';
```

File: front/tickettemplatemandatoryfield.form.php [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+$itiltype = 'Ticket';
+$fieldtype . 'Mandatory';
```
Comment (cedric-anne) [inline front/tickettemplatemandatoryfield.form.php]:
```suggestion
$fieldtype = 'Mandatory';
```

File: front/tickettemplatepredefinedfield.form.php [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+$itiltype = 'Ticket';
+$fieldtype . 'Predefined';
```
Comment (cedric-anne) [inline front/tickettemplatepredefinedfield.form.php]:
```suggestion
$fieldtype = 'Predefined';
```

File: .travis.yml [hunk 10]
```diff
@@ -64,15 +64,6 @@ cache:
   directories:
     - $HOME/.composer/cache
 
-notifications:
```
[the remark this answers is not in the document]
Comment (trasher) [reply .travis.yml]: Indeed, but I'll keep until PR has been validated

File: install/mysql/glpi-empty.sql [hunk 11]
```diff
@@ -2968,7 +2968,8 @@ CREATE TABLE `glpi_entities` (
   UNIQUE KEY `unicity` (`entities_id`,`name`),
   KEY `entities_id` (`entities_id`),
   KEY `date_mod` (`date_mod`),
-  KEY `date_creation` (`date_creation`)
+  KEY `date_creation` (`date_creation`),
+  KEY `tickettemplates_id` (`tickettemplates_id`)
```
[the remark this answers is not in the document]
Comment (trasher) [reply install/mysql/glpi-empty.sql]: This is a foreign key

**Comment by cedric-anne** [comment]
I noticed that `ITILCategory` object is named `Ticket category`. So on Problem/Change template tabs, you have a `Ticket category` tab.
image

File: install/update_94_95.php [hunk 12]
```diff
@@ -257,6 +257,80 @@ function update94to95() {
...
+         );
+      }
+
+      if (!$DB->tableExists("glpi_{$itiltype}templatehiddenfields")) {
+         $query = "CREATE TABLE `glpi_{$itiltype}templatehiddenfields` (
+            `id` int(11) NOT NULL AUTO_INCREMENT,
+            `{$itiltype}templates_id` int(11) NOT NULL DEFAULT '0',
+            `num` int(11) NOT NULL DEFAULT '0',
+            PRIMARY KEY (`id`),
+            UNIQUE KEY `unicity` (`{$itiltype}templates_id`,`num`)
```
Comment (cedric-anne) [inline install/update_94_95.php]: Other `*template*fields` have this key.
```suggestion
            UNIQUE KEY `unicity` (`{$itiltype}templates_id`,`num`),
            KEY `{$itiltype}templates_id` (`{$itiltype}templates_id`)
```

**Pull request merged**